builds.js

```diff
@@ -9,112 +13,313 @@ const environments = new Map([
+       * @private
+       */
+      this._cacheRefreshLimit = options.buildCache.refreshLimit || defaultCacheRefreshLimit;
+
+      /**
+       * @member {number} _cacheRefreshInterval The interval in ms on which the cache is refreshed refreshed
+       * @private
+       */
+      this._cacheRefreshInterval = (options.buildCache.refreshInterval || options.buildCache.refreshInterval === 0) ?
+        options.buildCache.refreshInterval :
+        defaultCacheRefreshInterval;
+      this.resumeCacheRefresh();
+    }
   }
 
-  //
-  // Handle scoped packages
-  //
-  pkg = encodeURIComponent(pkg);
+  /**
+   * Clears the build cache if one exists
+   *
+   * @public
+   */
+  clearCache() {
+    if (this._cache) {
+      this._cache.clear();
+    }
+  }
 
-  debug('Build metadata: pkg = %s, env = %s, version = %s', pkg, env, version);
-  return this.warehouse.send(
-    ['builds'].concat(pkg, env, version, meta).filter(Boolean).join('/'),
-    { query: { locale }},
-    fn
-  );
-};
+  /**
+   * Allows you to resume the process of refreshing the build cache
+   *
+   * @public
+   */
+  resumeCacheRefresh() {
+    if (this._cacheRefreshIntervalId === null && this._cacheRefreshInterval > 0) {
+      this._cacheRefreshIntervalId = setInterval(
+        this._refreshCache,
+        this._cacheRefreshInterval);
+    }
+  }
 
-/**
- * Get the set of build heads for the given parameters
- *
- * @param {Object} params Parameters that specify, env and pkg
- * @param {Function} fn Completion callback
- * @returns {Warehouse} fluent interface
- * @api public
- */
-Builds.prototype.heads = function heads(params, fn) {
-  const env = environments.get(params.environment || params.env) || 'dev';
-  const name = params.package || params.pkg;
+  /**
+   * Allows you to stop the process of refreshing the build cache
+   *
+   * @public
+   */
+  stopCacheRefresh() {
+    if (this._cacheRefreshIntervalId !== null) {
+      clearInterval(this._cacheRefreshIntervalId);
+      this._cacheRefreshIntervalId = null;
+    }
+  }
+
+  /**
+   * Refreshes the build cache.
+   *
+   * @private
+   */
+  _refreshCache() {
+    if (!this._cache) {
+      return;
+    }
+
+    if (this._cacheRefreshing) {
+      debug('Skipping a build cache refresh interval, previous interval took too long.');
+      return;
+    }
+
+    this._cacheRefreshing = true;
+
+    async.eachLimit(this._cache.values(), this._cacheRefreshLimit, (build, next) => {
+      this._get(build, error => {
+        if (error) {
+          const { pkg, env, version } = build;
+          debug('Error refreshing cache for: pkg = %s, env = %s, version = %s', pkg, env, version);
+          // Swallow cache refresh error, continue serving up the stale data as its better than nothing
+        }
+        next(null);
+      });
+    }, error => {
+      this._cacheRefreshing = false;
+      debug('Error refreshing cache: %s', error);
```

Comment:
Is this outputting every time? Since you're never returning the error within the callback above, `error` will always be null. So you don't likely need this debug statement.

Reply:
Ah right, will remove it.  It should technically never happen.